File: src/Monitor.App/Repositories/AuditRepository.cs
using Dapper;
using Monitor.App.Infrastructure.Database;
using Monitor.App.Models;

namespace Monitor.App.Repositories;

public class AuditRepository : IAuditRepository
{
    private readonly DbConnectionFactory _factory;

    public AuditRepository(DbConnectionFactory factory) => _factory = factory;

    public void LogAction(string action, string? details = null)
    {
        using var conn = _factory.CreateConnection();
        conn.Execute("INSERT INTO audit_log (action, details) VALUES (@Action, @Details)",
            new { Action = action, Details = details });
    }

    public List<AuditLog> GetAuditLogs(int limit = 100)
    {
        using var conn = _factory.CreateConnection();
        return conn.Query<AuditLog>(
            "SELECT id AS Id, action AS Action, details AS Details, timestamp AS Timestamp FROM audit_log ORDER BY timestamp DESC LIMIT @Limit",
            new { Limit = limit }).AsList();
    }
}

File: legacy/python-monitor/app/domain/repositories/audit_repository.py
"""
Audit log repository — handles read/write for `audit_log` table.
"""
from app.infrastructure.database.connection import ConnectionManager


class AuditRepository:
    """Data access for audit_log table."""

    def __init__(self, conn_mgr: ConnectionManager):
        self._conn_mgr = conn_mgr

    def log_action(self, action: str, details: str = None):
        """Write an audit log entry."""
        with self._conn_mgr.connection() as conn:
            cursor = conn.cursor()
            cursor.execute(
                "INSERT INTO audit_log (action, details) VALUES (?, ?)",
                (action, details)
            )
            conn.commit()

    def get_audit_logs(self, limit: int = 100) -> list:
        """Get recent audit log entries."""
        with self._conn_mgr.connection() as conn:
            cursor = conn.cursor()
            cursor.execute(
                "SELECT * FROM audit_log ORDER BY timestamp DESC LIMIT ?",
                (limit,)
            )
            return [dict(row) for row in cursor.fetchall()]
